lib/definition/component-info.ts

```diff
@@ -33,6 +38,12 @@ export function processInfo(componentsDefinition: ComponentsDefinition): Compone
             result[component.name].fields.forEach((f) => addRestrictChildrenInfo(componentsDefinition, component, f));
             return result;
         }, {} as ComponentInfoFields),
+        componentProperties: componentsDefinition.componentProperties.reduce((result, property) => {
```

Comment:
I'm not sure this is good enough. The full definition parses properties per component:
https://github.com/WoodWing/studio-component-set-tools/blob/master/lib/parser/parser-utils.ts#L130

It might be possible to override the dataType in the component property.

Reply:
It was (kind of) working as overriding the dataType is (currently) not allowed. But you are obviously right that this is not the correct approach so I rewrote it by using the parsed component set and specifying the properties per component.